receiver/hostmetricsreceiver/internal/metadata/generated_metrics.go

```diff
@@ -24,7 +24,7 @@ import (
 // Type is the component type name.
 const Type configmodels.Type = "hostmetricsreceiver"
 
-type metricIntf interface {
+type Metric interface {
```

Comment:
1. Does this need to be exported? (It was not previously).
2. Is it possible to come up with a less generic name? Looking at the name `Metric` I have a hard time understanding what it is supposed to do.

Reply:
@tigrannajaryan I had changed it to exported to be able to pass through the interface so the Init() function could be called generically - https://github.com/open-telemetry/opentelemetry-collector/blob/db35667fe72095b0585e1ca4edc9767f4ddbcdbe/receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go#L69-L76 
I think pattern will occur more as more of the metrics in the receiver are refactored to be based off the generated metrics. The previous use cases were only for the scrapers that emitted a single metric.

I'm indifferent towards the name - `MetricIntf`? `MetricFactory`? `GeneratedMetric`?

Reply:
OK, understood re exporting.

As for the name, please use something that reflects the purpose. I do not quite understand what it does to suggest a good name. The generic name `Metric` does not help me to understand what it is doing.

If unsure keep it unchanged.

Reply:
I reverted the name back to `MetricIntf` for now and added a comment above to describe it.